File: slack_emoji_exporter.php
<?php
$url = "https://slack.com/api/emoji.list?pretty=1&token=";
$token = "";
$res = json_decode(file_get_contents($url.$token), true);

    foreach($res['emoji'] as $key => $val) {
        $fileName = $key.'.png';

        echo "downloading ".$key."\n";
        $data = file_get_contents($val);
        file_put_contents('./'.$fileName, $data);
    }

File: slack_emoji_exporter.rb
require 'net/http'
require 'json'
require 'open-uri'

TOKEN = ""
URL = "https://slack.com/api/emoji.list?pretty=1&token=" + TOKEN

res = Net::HTTP.get(URI.parse(URL))
JSON.parse(res)["emoji"].emojiList.each {|k, v|
  open(URI.parse(v)) do |file|
    File.open(k + ".png", "wb") do |out|
      out.write(file.read)
    end
  end
}
